frontend/libs/studio-components/src/components/StudioCodelistEditor/StudioCodeListEditor.test.tsx

```diff
@@ -506,6 +502,105 @@ describe('StudioCodeListEditor', () => {
   });
 
   describe('Type handling', () => {
+    it('Renders type selector when code list is empty', () => {
+      renderCodeListEditor({ codeList: [] });
+      const typeSelector = screen.getByRole('combobox', {
+        name: texts.typeSelector.label,
+      });
+      expect(typeSelector).toBeInTheDocument();
+    });
+
+    it('Does not render type selector when code list is already populated', () => {
+      renderCodeListEditor();
+      const typeSelector = screen.queryByRole('combobox');
+      expect(typeSelector).not.toBeInTheDocument();
+    });
+
+    it("Creates an empty string item when string is selected and 'Add new' is pressed", async () => {
+      const user = userEvent.setup();
+      renderCodeListEditor({ codeList: [] });
+
+      const typeSelector = screen.getByRole('combobox');
```

Comment:
Can we add a name for this too? 😄 
```tsx
      const typeSelector = screen.getByRole('combobox', { name: xxx });
```

Reply:
I have created a function for this, to avoid duplicates.